File: Project/app/src/main/java/com/helpme/slideAdapter.java
package com.helpme;

import android.content.Context;
import android.content.Intent;
import android.graphics.Color;
import android.graphics.Typeface;
import android.support.v4.view.PagerAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import java.util.zip.Inflater;

public class slideAdapter extends PagerAdapter {

    private Context context;
    private LayoutInflater layoutInflater;
    private Typeface typeface;
    private String CarpenterDescription = "Our Program includes a lot of Carpenters that they can help you fix anything you need in the field of wood or you can invite them";
    private String PlumberDescription = "Our Program includes a lot of Plumbers that they can help you fix anything you need in the bathrooms or you can invite them";
    private String MechanicDescription = "Our Program includes a lot of Mechanics that they can help you fix anything you need in your car or you can invite them";


    // ourList of Images
    public int[] imgs = {R.drawable.carpenter, R.drawable.plumber,R.drawable.mechanic};
    public String[] Titles = {"Carpenter", "Plumber", "Mechanic"};
    public String[] Description = {CarpenterDescription, PlumberDescription, MechanicDescription};
    public int[] bkgnd = {
            R.drawable.gradient,
            R.drawable.gradient,
            R.drawable.gradient,
    };


    public slideAdapter(Context context) {
        this.context = context;
    }

    @Override
    public int getCount() {
        return imgs.length;
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {

        layoutInflater = (LayoutInflater) context.getSystemService(context.LAYOUT_INFLATER_SERVICE);
        View view = layoutInflater.inflate(R.layout.slide, container, false);
        LinearLayout layoutSlide = (LinearLayout) view.findViewById(R.id.slideLinearLayout);
        ImageView img = (ImageView) view.findViewById(R.id.slideImg);
        TextView tilte = (TextView) view.findViewById(R.id.titleSlide);
        TextView description = (TextView) view.findViewById(R.id.textDesciption);
        GUI gui = new GUI(this.context);
        GUI.changeTypeFace(tilte);

        tilte.setTextColor(Color.rgb(255, 255, 255));
        description.setTextColor(Color.rgb(255, 255, 255));
        layoutSlide.setBackgroundResource(bkgnd[position]);
        img.setImageResource(imgs[position]);
        tilte.setText(Titles[position]);
        description.setText(Description[position]);
        container.addView(view);
        return view;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        container.removeView((LinearLayout) object);
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return (view == (LinearLayout) object);
    }

}

File: Project/app/src/main/java/com/helpme/SlideAdapter.kt
package com.helpme

import android.content.Context
import android.graphics.Color
import android.graphics.Typeface
import android.support.v4.view.PagerAdapter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView


class SlideAdapter(var context: Context) : PagerAdapter() {

    private var CarpenterDescription: String = "Our Program includes a lot of Carpenters that they can help you fix anything you need in the field of wood or you can invite them";
    private var PlumberDescription: String = "Our Program includes a lot of Plumbers that they can help you fix anything you need in the bathrooms or you can invite them";
    private var MechanicDescription: String = "Our Program includes a lot of Mechanics that they can help you fix anything you need in your car or you can invite them";

    private val imgs = arrayListOf<Int>(R.drawable.carpenter, R.drawable.plumber, R.drawable.mechanic)
    private val Titles = arrayListOf<String>("Carpenter", "Plumber", "Mechanic")
    private val Description = arrayListOf<String>(CarpenterDescription, PlumberDescription, MechanicDescription)
    private val bkgnd = arrayListOf<Int>(R.drawable.gradient, R.drawable.gradient, R.drawable.gradient)


    override fun instantiateItem(container: ViewGroup, position: Int): Any {

        val layoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val view = layoutInflater.inflate(R.layout.slide, container, false)
        val layoutSlide = view.findViewById<LinearLayout>(R.id.slideLinearLayout)
        val img = view.findViewById<ImageView>(R.id.slideImg)
        val title = view.findViewById<TextView>(R.id.titleSlide)
        val description = view.findViewById<TextView>(R.id.textDesciption)

        title.typeface = Typeface.createFromAsset(context.assets, "Fonts/Nabila.ttf");
        title.setTextColor(Color.rgb(255, 255, 255))
        description.setTextColor(Color.rgb(255, 255, 255))
        layoutSlide.setBackgroundResource(bkgnd[position])
        img.setImageResource(imgs[position])
        title.text = Titles[position]
        description.text = Description[position]
        container.addView(view)
        return view
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(`object` as LinearLayout)
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view == `object` as LinearLayout
    }

    override fun getCount(): Int {
        return imgs.size
    }
}